doc/computing/randomness.rst

```diff
@@ -0,0 +1,345 @@
+splits::
+
+    >>> from sklearn.model_selection import KFold
+    >>> import numpy as np
+
+    >>> X = y = np.arange(10)
+    >>> rng = np.random.RandomState(0)
+    >>> cv = KFold(n_splits=2, shuffle=True, random_state=rng)
+
+    >>> for train, test in cv.split(X, y):
+    ...     print(train, test)
+    [0 3 5 6 7] [1 2 4 8 9]
+    [1 2 4 8 9] [0 3 5 6 7]
+
+    >>> for train, test in cv.split(X, y):
+    ...     print(train, test)
+    [0 4 6 7 8] [1 2 3 5 9]
+    [1 2 3 5 9] [0 4 6 7 8]
+
+We can see that the splits are different from the second time `split` is
+called. This may lead to unexpected results if you compare the performance of
+multiple estimators by calling `split` many times, as we will see in the next
+section.
+
+Common pitfalls and subtleties
+------------------------------
+
+While the rules that govern the `random_state` parameter are seemingly simple,
+they do however have some subtle implications. In some cases, this can even
+lead to wrong conclusions.
+
+Estimators
+..........
+
+**Different `random_state` types lead to different cross-validation
+procedures**
+
+Depending on the type of the `random_state` parameter, estimators will behave
+differently, especially in cross-validation procedures. Consider the
+following snippet::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import cross_val_score
+    >>> import numpy as np
+
+    >>> X, y = make_classification(random_state=0)
+
+    >>> rf_123 = RandomForestClassifier(random_state=123)
+    >>> cross_val_score(rf_123, X, y)
+    array([0.85, 0.95, 0.95, 0.9 , 0.9 ])
+
+    >>> rf_inst = RandomForestClassifier(random_state=np.random.RandomState(0))
+    >>> cross_val_score(rf_inst, X, y)
+    array([0.9 , 0.95, 0.95, 0.9 , 0.9 ])
+
+We see that the cross-validated scores of `rf_123` and `rf_inst` are
+different, as should be expected since we didn't pass the same `random_state`
+parameter. However, the difference between these scores is more subtle than
+it looks, and **the cross-validation procedures that were performed by**
+:func:`~sklearn.model_selection.cross_val_score` **significantly differ in
+each case**:
+
+- Since `rf_123` was passed an integer, every call to `fit` uses the same RNG:
+  this means that all random characteristics of the random forest estimator
+  will be the same for each of the 5 folds of the CV procedure. In
+  particular, the (randomly chosen) subset of features of the estimator will
+  be the same across all folds.
+- Since `rf_inst` was passed a `RandomState` instance, each call to `fit`
+  starts from a different RNG. As a result, the random subset of features
+  will be different for each folds.
+
+While having a constant estimator RNG across folds isn't inherently wrong,
+results are more significant if we allow the estimator RNG to vary for each
+fold, so passing an instance may be preferable (more on this later).
```

Comment:
```suggestion
results characterize best the learner if we allow the estimator RNG to vary for each
fold, hence passing an instance may be preferable (more on this later).
```
I would avoid the use of the terminology "significant", which requires specifying a null hypothesis.

Reply:
Ok, I'll use "robustness w.r.t. randomness" instead of significance.